### ISSUE TITLE
Can you add a feature to capture specific network requests? 

### ISSUE THREAD

**Issue by chaoqunxie** [Skyvern-AI/skyvern#1190]
Can you add a feature to capture specific network requests? I need to get information from the network request headers. Also, it seems the F12 developer tools and bookmark creation are not supported.

**Comment by chaoqunxie** [comment]
Open the browser's developer tools, navigate to the console tab, and execute the JavaScript command: `localStorage.getItem('userToken')`. 

1. Log in to your account:
   - Enter your phone number in the phone input field
   - Enter your password in the password field
   - Check the "I have read and agree" checkbox
   - Click the login button

2. After successful login, follow these steps to open Developer Tools:
   2.1 Press F12 key to open Developer Tools
       - If F12 doesn't work, try pressing Fn + F12
       
   2.2 Find and click "Console" tab:
       - Look at the top or bottom of Developer Tools window
       - Find the tab labeled "Console"
       - Click on "Console" tab
       
   2.3 Click the console command input area:
       - Look for the input area at bottom of Console panel
       - Click where you see the `>` symbol or blinking cursor
       
   2.4 Type or paste this command:
       localStorage.getItem('userToken')
       
   2.5 Press Enter key to execute the command

3. The userToken value will be shown in the console output area.

error msg:

"Terminate because the platform does not provide options to open developer tools or execute JavaScript from the console on this page."
![image](https://github.com/user-attachments/assets/1ba3e202-974d-4584-855e-097d127ba06e)

**Comment by suchintan** [comment]
Hey! We already have this feature :) 

You can grab all of the HAR logs of a particular Skyvern session out as an artifact, which contains all network calls 

I need to add this to our public documentation

https://github.com/Skyvern-AI/skyvern/blob/3d6c1dd60c5497d18d9ae9226fcc2ac94ce6ecd2/skyvern/forge/sdk/artifact/models.py#L43


@base_router.get(
    "/tasks/{task_id}/steps/{step_id}/artifacts",
    tags=["agent"],
    response_model=list[Artifact],
)
@base_router.get(
    "/tasks/{task_id}/steps/{step_id}/artifacts/",
    tags=["agent"],
    response_model=list[Artifact],
    include_in_schema=False,
)
async def get_agent_task_step_artifacts(
    task_id: str,
    step_id: str,
    current_org: Organization = Depends(org_auth_service.get_current_org),
) -> Response:

**Comment by chaoqunxie** [comment]
Thank you very much for your reply. I think you misunderstood my intention. I want to directly output the final result I need through this project: this result refers to network requests of a specific website, not network requests between large language models. The information I've obtained through /api/v1/tasks/tsk_326033608332124018/steps/stp_326033608332124020/artifacts is almost entirely about requests between large language models. Even if I want to obtain these requests, I don't want to do so through this API method, but rather directly use the large language model to obtain them. It's difficult to obtain task_id and step_id through the API method.

**Comment by suchintan** [comment]
We do store the network requests as HAR artifacts and can be found under here: `export HAR_PATH=/data/har`

**Comment by suchintan** [comment]
The HAR file contains all network traffic (not related to LLMs)

**Comment by chaoqunxie** [comment]
You're correct. Currently, finding the corresponding HAR file through the task_id and manually importing it into the browser to view network request details is a feasible method, but it's not intuitive or efficient. I'm very interested in exploring how to directly retrieve and display network request details within Skyvern. This would significantly enhance debugging and analysis convenience.

**Comment by suchintan** [comment]
Ah! I did misinterpret your request. I assumed you wanted to use it for "post analysis"

We will take this feature request. We want to expose 3 things to our users:

1. The raw Skyvern logs (which you can see via `docker logs`)
2. The browser console logs 
3. The browser network requests (incl error codes if there may be any)

Great idea.
